README.md

```diff
@@ -177,44 +177,65 @@ Example of a partial CR:
 
 ### Mounting and injecting binding information
 
-This specification allows for data to be mounted using volumes or injected using environment variables.  The best practice is to mount any sensitive information, such as passwords, since that will avoid accidentally exposure via environment dumps and subprocesses.  Also, binding binary data (e.g. .p12 certificate for Kafka) as an environment variable might cause a pod to fail to start (stuck on `CrashLoopBackOff`), so it advisable for backing services with such binding data to mark it with `bindAs: volume`.
+#### Mounting data
+Implementations of this specification **MUST** mount the binding data into the consuming application container in the following location:
+
+```
+<mountPathPrefix>/<service-id>/secret/<persisted_secret>
+```
+
+Where:
+* `<mountPathPrefix>` defaults to `/platform/bindings` if not specified in the `ServiceBinding` CR via the `mountPathPrefix` element.  The environment variable `SERVICE_BINDINGS_ROOT` **MUST** be set to the value of `<mountPathPrefix>`, so that applications can always find this directory, and it is immutable.  This means that if another `ServiceBinding` CR wants to project itself into the same container, it **MUST** reuse the current `SERVICE_BINDINGS_ROOT` value.  
+* `<service-id>` is the `id` field of the corresponding `service` entry in the `ServiceBinding` CR.  If the `id` field is not present, the `name` field is used instead.  The `<service-id>` path **MUST** be unique between the services bound to a particular application.
+* `<persisted_secret>` represents a set of files where the filename is a Secret key and the file contents is the corresponding value of that key.
+
+Example:  `/platform/bindings/mongo-db/secret/MONGODB_HOST`
+
+#### Cross services data mappings
+
+The `ServiceBinding` CR allows for the declaration of mappings that are potentially composed of different service bindings (ids).  For this case, the special id of `dataMappings` is reserved and **MUST NOT** be used by any service bindings.  
```

Comment:
this is a comment from implementation perspective, right now in the `ServiceBinding` CR,  `customEnvVar` is global for all backing services, what if we have `customEnvVar` per backing service ? that way we can completely avoid the scenario of cross services data mappings and unnecessary complexity for the application to go to multiple paths to get the binding information of one single service.

From this
```
backingServiceSelectors:
    - group: postgres.dev
      kind: Service
      resourceRef: user-db
      version: v1beta1
      id: X
    - group: postgres.dev
      kind: Credentials
      resourceRef: admin-creds
      version: v1beta1
      id: Y
customEnvVar:
    - name: CONNECTION_URL
       value: {{  X.status.username }} / {{ Y.status.url }}
```

to this 

```
backingServiceSelectors:
    - group: postgres.dev
      kind: Service
      resourceRef: user-db
      version: v1beta1
      id: X
    - group: postgres.dev
      kind: Credentials
      resourceRef: admin-creds
      version: v1beta1
      id: Y
      customEnvVar:
         - name: CONNECTION_URL
         value: {{  X.status.username }} / {{ Y.status.url }}
```
Service binding CR yaml snippet extracted from here https://github.com/redhat-developer/service-binding-operator/issues/396

Thoughts @arthurdm @navidsh @sbose78 ?

Reply:
@yharish991 Yea, I think that would simplify things a lot but then there is this concern:
https://github.com/redhat-developer/service-binding-operator/issues/356#issuecomment-596049739

`CONNECTION_URL` is technically relying on info from service (`id: X`) but is defined under another service. But in reality, it relies on both services to be defined in the `ServiceBinding` CR.

Reply:
ya.  I think it's a good thought @yharish991 , but as @navidsh pointed out it is counter intuitive to have something under one service relying on another.  I modified the PR to call the special id `custom-mappings`, so it's very clear this is something custom / composed by the requester.

Reply:
In SQl, its quite common to refer to multiple tables all over the query, so was thinking why not here, the developers are already used to this kind of thinking, so i think it would not be a problem to them, also it solves the problem of identifying the binding info if we use `custom-mappings` as the special id. For example if the directory `<mountPathPrefix>/custom-bindings/secret/` has file `CONNECTION_URL `, how do we identify which service it belongs to?

Reply:
I think that when a custom mapping is used, if it refers to a particular service it should use a reasonable prefix?